includes/admin/load.php

```diff
@@ -262,79 +292,92 @@ function perflab_activate_plugin() {
 		$plugin            = $plugin . '/' . $plugin_file_names[0];
 	}
 
-	if ( ! current_user_can( 'activate_plugin', $plugin ) ) {
-		wp_die( esc_html__( 'Sorry, you are not allowed to activate this plugin.', 'default' ) );
+	$result = activate_plugin( $plugin );
+	if ( is_wp_error( $result ) ) {
+		wp_die( esc_html( $result->get_error_messages() ) );
 	}
 
-	// Activate the plugin in question and return to prior screen.
-	$do_plugin_activation = activate_plugins( $plugin );
-	$referer              = wp_get_referer();
-	if ( ! is_wp_error( $do_plugin_activation ) ) {
-		$referer = add_query_arg(
-			array(
-				'activate' => true,
-			),
-			$referer
-		);
-	}
+	$url = add_query_arg(
+		array(
+			'page'     => PERFLAB_SCREEN,
+			'activate' => 'true',
+		),
+		admin_url( 'options-general.php' )
+	);
 
-	if ( wp_safe_redirect( $referer ) ) {
+	if ( wp_safe_redirect( $url ) ) {
 		exit;
 	}
 }
-add_action( 'admin_action_perflab_activate_plugin', 'perflab_activate_plugin' );
 
 /**
- * Callback function hooked to admin_action_perflab_deactivate_plugin to handle plugin deactivation.
+ * Callback function to handle admin inline style.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_deactivate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
+function perflab_print_features_page_style() {
+	?>
+<style type="text/css">
+	.plugin-card .name, .plugin-card .desc {
+		margin-left: 0;
 	}
-
-	// The plugin being deactivated.
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
-
-	check_admin_referer( "perflab_deactivate_plugin_{$plugin}" );
-
-	if ( ! current_user_can( 'deactivate_plugin', $plugin ) ) {
-		wp_die( esc_html__( 'Sorry, you are not allowed to deactivate this plugin.', 'default' ) );
+	.plugin-card-top {
+		min-height: auto;
 	}
+</style>
+	<?php
+}
 
-	// Deactivate the plugin in question and return to prior screen.
-	deactivate_plugins( $plugin );
-	$referer = add_query_arg(
-		array(
-			'deactivate' => true,
-		),
-		wp_get_referer()
-	);
-	if ( wp_safe_redirect( $referer ) ) {
-		exit;
-	}
+/**
+ * Callback function that print plugin activation script.
+ *
+ * @since n.e.x.t
+ */
+function perflab_print_plugin_activation_script() {
+	$js = <<<JS
+( function ( document ) {
+	document.addEventListener( 'DOMContentLoaded', function () {
+		document.addEventListener( 'click', function ( event ) {
+			if (
+				event.target.classList.contains(
+					'perflab-install-active-plugin'
+				)
+			) {
+				const target = event.target;
+				target.classList.add( 'updating-message' );
+				target.innerHTML = wp.i18n.__( 'Activating…', 'performance-lab' );
```

Comment:
WordPress doesn't (yet) support JS translations for inline scripts.

So right now there is no way to translate this string.

A few options:

1. Put this back into an external file and call `wp_set_script_translations()`
2. Register a script handle and call `wp_set_script_translations()`, but continue printing as inline script
3. Print this in PHP instead and use as variable here (e.g. `$activating = esc_js( ... );` at the top of the function and then use `$activating` here.

Reply:
Make sense. Removed the JS.